engine/src/main/java/org/hibernate/search/backend/impl/DeleteByQuerySupport.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.NumericUtils;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.backend.SingularTermQuery;
+import org.hibernate.search.backend.SingularTermQuery.Type;
+import org.hibernate.search.exception.AssertionFailure;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * This class has means to convert all (by default) supported DeletionQueries back to Lucene Queries and to their
+ * String[] representation and back.
+ *
+ * @author Martin Braun
+ */
+public final class DeleteByQuerySupport {
+
+	private DeleteByQuerySupport() {
+		throw new AssertionError( "can't touch this!" );
+	}
+
+	public static final Map<Integer, ToLuceneQuery> TO_LUCENE_QUERY_CONVERTER;
```

Comment:
About the parse functions:

These are only used for serialization right now. I kinda don't want these in the serialization modules because they are needed in both and I don't want to spread the DeleteByQuery logic in too many modules.